File: BEGINNER/Python/1040.py
# -*- coding: utf-8 -*-

N1, N2, N3, N4 = map(float, input().split())

nota_exame = 0.0

media = (((N1 * 2) + (N2 * 3) + (N3 * 4) + (N4 * 1)))/10

print("Media: %0.1f" %media)

if   (media >= 7.0):
    print("Aluno aprovado")
elif (media < 5.0):
    print("Aluno reprovado")
    nota_exame = float(input())
    print("Nota do exame: %0.1f" %nota_exame)
    media = (media + nota_exame)/2
elif (media >= 5.0 and media <= 6.9):
    print("Aluno em exame")
print("Media final: %0.1f" %media)

File: BEGINNER/CPlusPlus/1040.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main () {
    
    double N1, N2, N3, N4, Nexame = 0.0;
    
    
    int P1 = 2;
    int P2 = 3;
    int P3 = 4;
    int P4 = 1;
    
    cin >> N1 >> N2 >> N3 >> N4;

    
    double media = ((N1 * P1) + (N2 * P2) + (N3 * P3) + (N4 * P4))/(P1 + P2 + P3 + P4);

    cout << fixed << setprecision(1);
    cout << "Media: " << media << endl;

    if (media >= 7.0){
        cout << "Aluno aprovado." << endl;
    }

    else if(media < 5.0){
        cout << "Aluno reprovado." << endl;
    }

    else if (media >= 5.0 && media <= 6.9){
        cout << "Aluno em exame." << endl;
        cin >> Nexame;
        double mediaFinal = (media + Nexame)/2;
        cout << "Nota do exame: " << Nexame << endl;
        cout << "Aluno aprovado." << endl;
        cout << "Media final: " <<  mediaFinal << endl;
    }

    else{
        cout << "Periodo --" << endl;
    }
return 0;    
}
